components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -123,24 +134,35 @@ func (s *WorkspaceService) ListWorkspaces(ctx context.Context, req *connect.Requ
 }
 
 func (s *WorkspaceService) UpdatePort(ctx context.Context, req *connect.Request[v1.UpdatePortRequest]) (*connect.Response[v1.UpdatePortResponse], error) {
+	workspaceID, err := validateWorkspaceID(req.Msg.GetWorkspaceId())
+	if err != nil {
+		return nil, err
+	}
+
 	conn, err := getConnection(ctx, s.connectionPool)
 	if err != nil {
 		return nil, err
 	}
-	if req.Msg.Port.Policy == v1.PortPolicy_PORT_POLICY_PRIVATE {
-		_, err = conn.OpenPort(ctx, req.Msg.GetWorkspaceId(), &protocol.WorkspaceInstancePort{
+
+	switch req.Msg.GetPort().GetPolicy() {
+	case v1.PortPolicy_PORT_POLICY_PRIVATE:
+		_, err = conn.OpenPort(ctx, workspaceID, &protocol.WorkspaceInstancePort{
 			Port:       float64(req.Msg.Port.Port),
 			Visibility: protocol.PortVisibilityPrivate,
 		})
-	} else if req.Msg.Port.Policy == v1.PortPolicy_PORT_POLICY_PUBLIC {
-		_, err = conn.OpenPort(ctx, req.Msg.GetWorkspaceId(), &protocol.WorkspaceInstancePort{
+	case v1.PortPolicy_PORT_POLICY_PUBLIC:
+		_, err = conn.OpenPort(ctx, workspaceID, &protocol.WorkspaceInstancePort{
 			Port:       float64(req.Msg.Port.Port),
 			Visibility: protocol.PortVisibilityPublic,
 		})
+	default:
+		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("Unknown port policy specified."))
 	}
 	if err != nil {
+		log.WithField("workspace_id", workspaceID).Error("Failed to update port")
 		return nil, proxy.ConvertError(err)
 	}
```

Comment:
Drive-by cleanup: this ensures that an unknown port policy fails on the API layer. Previously, it would skip both branches and return OK, without actually doing anything.